tests/integration/profiling/rule_based_profilers/test_run_profiler_notebook.py

```diff
@@ -46,10 +46,17 @@ def test_run_rbp_notebook(tmp_path):
             nbformat.write(nb, f)
 
     # clean up Expectations directory after running test
-    shutil.rmtree(os.path.join(base_dir, "great_expectations/expectations/tmp"))
-    os.remove(
-        os.path.join(base_dir, "great_expectations/expectations/.ge_store_backend_id")
-    )
+    try:
+        shutil.rmtree(os.path.join(base_dir, "great_expectations/expectations/tmp"))
+        os.remove(
+            os.path.join(
+                base_dir, "great_expectations/expectations/.ge_store_backend_id"
+            )
+        )
+    except FileNotFoundError:
+        # This just means the files were already deleted by running the optional last cell in the notebook.
+        # Therefore we allow the test to pass.
+        pass
```

Comment:
I think this file deletion should occur nested under the `finally` block - if anything goes wrong earlier on in the test, these files will not be deleted.

Also, do we need to write the results to a temp notebook? What are your thoughts on devs just running the original notebook to debug?

Finally, could we please log when we hit the `FileNotFoundError` so that comment you wrote is presented in stdout?

Reply:
Great suggestions.I thought the point on adding the `FileNotFoundError` to the log was fantastic, so my most recent commit has that change. I have some questions about what you had in mind for `finally` block and was wondering if you had time to discuss tomorrow?